src/targets/maven.ts

```diff
@@ -102,48 +93,115 @@ export class MavenTarget extends BaseTarget {
     );
   }
 
-  private getEnvVarOrDefault(envVar: string): string {
+  private getEnvVarOrDefault(envVar: OptionalConfig): string {
     if (process.env[envVar]) {
       return process.env[envVar] as string; // `as string` to make TS happy
     }
     return DEFAULT_ENV_VARIABLES[envVar];
   }
 
-  public async publish(version: string, _revison: string): Promise<void> {
-    await withTempDir(
-      async dir => {
-        const git = simpleGit(dir);
-        const username = await getAuthUsername(this.github);
-        this.githubRemote.setAuth(username, getGithubApiToken());
-        await git.clone(this.githubRemote.getRemoteStringWithAuth(), dir);
-        await git.checkout(`release/${version}`); // TODO: release name should be customized
-
-        await this.createUserGradlePropsFile();
-        const mavenReleaser = new MavenReleaser(
-          this.mavenConfig.distributionsPath,
-          this.mavenConfig.settingsPath,
-          this.mavenConfig.mavenRepoUrl,
-          this.mavenConfig.mavenRepoId,
-          this.mavenConfig.mavenCliPath,
-          this.mavenConfig.gradleCliPath
-        );
-        mavenReleaser.release();
-
-        await git.add(FILES_TO_COMMIT);
-        await git.commit(`craft(maven): Deployed ${version} to Maven Central.`);
-        if (this.shouldPush()) {
-          await git.push();
+  public async publish(_version: string, _revison: string): Promise<void> {
+    this.createUserGradlePropsFile();
+    if (isDryRun()) {
+      this.logger.info('[dry-run] Not uploading to Maven.');
+      return;
+    }
+    this.upload();
+    this.closeAndRelease();
+  }
+
+  /**
+   * Deploys to Maven Central the distribution packages.
+   * Note that after upload, this must be `closeAndRelease`.
+   */
+  public upload(): void {
+    const distributionsDirs = fs.readdirSync(
+      this.mavenConfig.distributionsPath
+    );
+    for (const distDir of distributionsDirs) {
+      const moduleName = path.parse(distDir).base;
+      const androidFile = this.getAndroidDistributionFile(distDir);
+      const targetFile = androidFile
+        ? androidFile
+        : path.join(distDir, `${moduleName}.jar`);
+      const javadocFile = path.join(distDir, `${moduleName}-javadoc.jar`);
+      const sourcesFile = path.join(distDir, `${moduleName}-sources.jar`);
```

Comment:
Where are we getting these from? These should be fetched from the artifact provider from the release build?

This is why you need the `version` and `revision` arguments in `publish`: to get the associated artifacts.

Reply:
This comment needs an answer :)

Reply:
all the files except the `.asc` files which are generated during the gpg signing should be downloaded from the artifact provider.